### ISSUE TITLE
import block for_each

### ISSUE THREAD

**Issue by paulaustin-automutatio** [opentofu/opentofu#1287]
### OpenTofu Version

```shell
OpenTofu v1.6.1
on windows_amd64
```

### Use Cases

Where resource blocks are defined using for_each you must currently manually add seperate import blocks for each of them if you want to be able to re-import resources if the state got corrupted.

### Attempted Solutions

N/A

### Proposal

Add support for for_each to the import block. This would work in the same way as for_each for the resource block in that the suffix is added to the resource reference.

The following 
```
import {
 for_each = { for n in  [
    "admin",
    "cosmos": n => n }
  to = azurerm_application_security_group.asg
  id = ".../asg${each.key}"
}
```

would be instead of

```
import {
  to = azurerm_application_security_group.asg["admin"]
  id = ".../asgadmin"
}

import {
  to = azurerm_application_security_group.asg["cosmos"]
  id = "..../asgcosmos"
}
```

**Comment by janosdebugs** [comment]
Thanks for opening this @paulaustin-automutatio we'll discuss this in the core team and get back to you.

**Comment by paulaustin-automutatio** [comment]
@janosdebugs FYI terraform added for_each to import for v 1.7.x. It behaves differently from for_each in resources

**Comment by RLRabinowitz** [comment]
Hey @paulaustin-automutatio .

There's already an issue for supporting `for_each` as part of the `import` block, and I'm currently working on it 😄 . It will be available in OpenTofu v1.7.0

See https://github.com/opentofu/opentofu/issues/1033
